### PR TITLE
Rebased breach resolution feature

### PR DESCRIPTION
**Pull request by groovecoder**
Combining https://github.com/mozilla/blurts-server/pull/1479 and https://github.com/mozilla/blurts-server/pull/1467 here. @lesleyjanenorton instructions from #1479 are best ...

WIP breach resolution feature updates so @groovecoder and I can get our Resolved Breach Ducks in a row. 🦆🦆🦆

🦆 `/dashboard` Updates: 
- If at least one breach has been resolved, the second and third "breach stats" at page top change to reflect the number of resolved passwords and the number of passwords still exposed in unresolved breaches.
- Previously resolved breaches are hidden by default and gated behind "Show Resolved" buttons.
- The CTA for unresolved breach cards in the dashboard is changed to "Resolve this breach"
- The CTA for resolved breach cards reverts back to "More about this breach"

Adds a progress bar to the dashboard and...
- If a user has at least one breach, an intro to resolving breaches is visible below the breach stats.
- If a user has resolved at least one breach, the intro is replaced with a progress bar that  indicates the number of resolved/unresolved breaches.
- If a user has resolved all of their breaches, the progress bar is replaced with some messaging telling the user they've resolved everything.

🦆 `/breach-details` Updates: 
- Adds a top of page notification if the user has an email that was exposed in that breach and the breach hasn't been marked as resolved.
- Adds list of all verified emails that were exposed in the breach and adds ability to mark them as resolved/unresolved
- Adds a confirmation modal with varying messaging and imagery based on the number of resolved breaches.

### CODE REVIEW

File: views/partials/affected-emails.hbs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ <div class="resolve-breaches-wrapper drop-shadow">
+  <h3 class="bold resolve-headline txt-cntr">{{ resolutionStrings.subhead }}</h3>
+  <div class="affected-emails-list-wrapper">
+    <p class="resolution-message">{{{ resolutionStrings.message }}}</p>
+    <ul class="affected-emails-list">
+      {{#each affectedEmails}}
+        <li class="flx space-between affected-email-list-item">
+          <p class="flx cntr demi affected-email">{{ emailAddress }}</p>
+            <button data-affected-email="{{ emailAddress}}" data-passwords-exposed="{{ ./../breachExposedPasswords }}" data-recency-index="{{ recencyIndex }}" data-is-resolved="{{ isResolved }}"
```
Comment (groovecoder) [inline views/partials/affected-emails.hbs]: @lesleyjanenorton I'm changing this to `data-is-already-resolved` to make the back-end logic more straight-forward.

Comment (groovecoder) [reply views/partials/affected-emails.hbs]: oops, nvm.

**Comment by pdehaan** [comment]
I have two accounts that are in the Adobe breach, but nothing seems to happen when I try and resolve either breach. DevTools console seems to say:

> Uncaught (in promise) ReferenceError: ga is not defined
> &nbsp;&nbsp; at c (app.min.js:1)
> &nbsp;&nbsp; at HTMLButtonElement.<anonymous> (app.min.js:1)

Not sure if this is due to me running some ad blocker or ghostery or some other thing that might be blocking some ga script.

**Comment by pdehaan** [comment]
Similarly, I get the "ga is not defined" error in the DevTools console when clicking the "What to do next" link in the top banner area.

monitor-v2 herokuapp com_breach-details_CafePress

**Comment by pdehaan** [comment]
Ref: #1497

**Comment by pdehaan** [comment]
I was able to resolve the 500px breach for one of my accounts without seeing the **ga** error, but the modal shortly afterwords was a bit... sparse?

2020-01-22__11_34_AM

**Comment by pdehaan** [comment]
Ref: #1508; "The image is not displayed for the first resolved breach on a newly created profile."

File: public/js/monitor.js [hunk 2]
```diff
@@ -92,16 +92,22 @@ function handleFormSubmits(formEvent) {
     email = thisForm.email.value.trim();
     thisForm.email.value = email;
   }
+  const formClassList = thisForm.classList;
```
Comment (groovecoder) [inline public/js/monitor.js]: This seems to work fine, but it reminds me when [we accidentally assigned an element's property value into a variable, manipulated the variable value, and forgot to assign the manipulated value back into the underlying element](https://github.com/mozilla/blurts-server/commit/85fd4e83c1a03b06143b6cc806f8974d676ca34a#diff-12528c522ce5f8325d894770704401fb).

To ensure the element's `classList` value is updated before submitting the form, can we add a line `thisForm.classList = formClassList` before `thisForm.submit()` ?

**Pull request merged**